Return paths to generated CDFs from post_processing

pleasant-menlo opened the pull request:
## Summary
Returns paths to CDFs written during ProcessInstrument post_processing.

## Testing
Added unit test verifying the path returned by write_cdf is returned by post_processing

- George and Ethan

Commit:
Return paths to generated CDFs from post_processing

greglucas commented on imap_processing/cli.py:
```diff
@@ -517,7 +517,7 @@ def post_processing(
         processed_data: list[xr.Dataset | Path],
         dependencies: ProcessingInputCollection,
-    ) -> None:
+    ) -> Optional[list[Path]]:
```
I don't love returning different values here. Could you change this to always return the list. I think if you move up `products = []` to the top, then in the early return case you can just `return products` there as well which would be an empty list.
```suggestion
    ) -> list[Path]:
```

pleasant-menlo replied:
Thanks for the callout! We've updated this and added corresponding tests in the second commit.

Ethan and George

maxinelasp reviewed (approved):
Looks good! I saw in your email you mentioned a failing codecov - sometimes the first codecov run fails and then once all the tests finish the re-run succeeds. We only have one test that uses all the remote data to run large/complicated tests, so the partial runs will fail the codecov. So, it looks like the PR is good as far as code coverage goes.

It does look like there are a few pre-commit bugs which should be fixed prior to merging.

greglucas reviewed (commented):
Seems fine to me 👍 , shouldn't really impact anything on our end since it was just returning None and we weren't doing anything with it.

Commit:
Return empty list from post_processing when no products to write

pleasant-menlo commented:
> Looks good! I saw in your email you mentioned a failing codecov - sometimes the first codecov run fails and then once all the tests finish the re-run succeeds. We only have one test that uses all the remote data to run large/complicated tests, so the partial runs will fail the codecov. So, it looks like the PR is good as far as code coverage goes.
> 
> It does look like there are a few pre-commit bugs which should be fixed prior to merging.

@maxinelasp - to clarify, are there bugs we are missing in our work and should address, or do you mean prior to this PR? Thanks!

maxinelasp commented:
@pleasant-menlo If you look here: https://results.pre-commit.ci/run/github/654679818/1753982459.OnCeqyTeTPyVBjRYxc09Zw

There are a few minor changes to docs, typehints, etc that should be fixed before we can merge. Once the "pre-commit.ci" check passes, we can merge this in.

Commit:
Address linter errors

Commit:
Address linter errors

Commit:
Address linter errors

Commit:
Address linter errors

pleasant-menlo commented:
> @pleasant-menlo If you look here: https://results.pre-commit.ci/run/github/654679818/1753982459.OnCeqyTeTPyVBjRYxc09Zw
> 
> There are a few minor changes to docs, typehints, etc that should be fixed before we can merge. Once the "pre-commit.ci" check passes, we can merge this in.

@maxinelasp Thanks for pointing us in the correct direction on this! Sorry for all of the extra commits while we addressed the issues...

- George and Ethan

maxinelasp commented:
No problem! We squash before merging so it won't be clear :) Plus, dealing with pre-commit can be challenging...

We're planning on making a release today so this change should be quickly available.
